src/web/app/components/copy-course-modal/copy-course-modal.component.ts

```diff
@@ -113,10 +115,11 @@ export class CopyCourseModalComponent implements OnInit {
   }
 
   /**
-   * Clears all selected feedback sessions.
+   * Clears all selected feedback sessions and fetch feedback sessions for chosen course.
    */
-  clearSelectedFeedbackSession(): void {
+  onSelectCourseChange(): void {
     this.selectedFeedbackSessions.clear();
```

Comment:
Another thing: should we consider the scenario where the user just plays around with courses, incurring additional session fetch?
Do you think it is safe to cache the fetched session locally or it will create a consistency problem?

Reply:
@moziliar I dont think we need to worry about the scenario where the user just plays around with courses, since most ppl wont do that :)

It is ok to cache, but the results might not always be up to date

Reply:
I see. Then it is ok to use server call for the latest data.

Reply:
> since most ppl wont do that

Be careful, assumptions like this are what evil people abuse.

You should cache. Look at the condition which will make the cached value outdated: someone else in the course, with sufficient permission, edits the feedback session at the same time.

Compare that condition (involves multiple parties, and the timing has to be exact) and people playing around/abusing the course selection (need just one person). I think you can agree which one has to be sacrificed.

Reply:
@wkurniawan07 Regarding updating the cache, I think before fetching the whole feedback sessions, we can check the `updatedAt` value then decide whether to make another call to get the data, or keep the current one. I dont think we have an API for the 1st check yet

So I'm still unsure whether updating the cache should be our responsibility

Reply:
Why do you need an API in the first place? There's no cache in the back-end and I never suggested that.

Reply:
> Look at the condition which will make the cached value outdated: someone else in the course, with sufficient permission, edits the feedback session at the same time.

I think I misunderstood this, but then I'm not sure how to invalidate the cache without calling the backend

Reply:
> how to invalidate the cache

Simple: reload the page. If you want to make it invalidated upon closing the modal, that also works.
I did admit that I used the term "cache" rather loosely, but you should get what I mean.